### ISSUE TITLE
IndexError: shape mismatch: indexing arrays could not be broadcast together with shapes (22361,3) (22361,) 

### ISSUE THREAD

**Issue by Violetit** [chrischoy/FCGF#38]
Hi @chrischoy @sjnarmstrong 
Thank you for sharing this wonderful work!
When I run the train.py script with 3dmatch dataset, something goes wrong, the error was report below:

```
09/26 12:59:16 Resetting the data loader seed to 0
Traceback (most recent call last):
  File "train.py", line 84, in <module>
    main(config)
  File "train.py", line 63, in main
    trainer.train()
  File "/disk/tia/FCGF/lib/trainer.py", line 124, in train
    val_dict = self._valid_epoch()
  File "/disk/tia/FCGF/lib/trainer.py", line 314, in _valid_epoch
    input_dict = data_loader_iter.next()
  File "/home/ubuntu/.conda/envs/py3-fcgf/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 363, in __next__
    data = self._next_data()
  File "/home/ubuntu/.conda/envs/py3-fcgf/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 989, in _next_data
    return self._process_data(data)
  File "/home/ubuntu/.conda/envs/py3-fcgf/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 1014, in _process_data
    data.reraise()
  File "/home/ubuntu/.conda/envs/py3-fcgf/lib/python3.7/site-packages/torch/_utils.py", line 395, in reraise
    raise self.exc_type(msg)
IndexError: Caught IndexError in DataLoader worker process 0.
Original Traceback (most recent call last):
  File "/home/ubuntu/.conda/envs/py3-fcgf/lib/python3.7/site-packages/torch/utils/data/_utils/worker.py", line 185, in _worker_loop
    data = fetcher.fetch(index)
  File "/home/ubuntu/.conda/envs/py3-fcgf/lib/python3.7/site-packages/torch/utils/data/_utils/fetch.py", line 44, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/home/ubuntu/.conda/envs/py3-fcgf/lib/python3.7/site-packages/torch/utils/data/_utils/fetch.py", line 44, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/disk/tia/FCGF/lib/data_loaders.py", line 252, in __getitem__
    pcd0.colors = o3d.utility.Vector3dVector(color0[sel0])
IndexError: shape mismatch: indexing arrays could not be broadcast together with shapes (22361,3) (22361,) 
```
Cloud you please help me to solve this problem?
Thank you very much!

**Comment by DiegoWangSys** [comment]
use sel0[1] instead of sel0

**Comment by Violetit** [comment]
> use sel0[1] instead of sel0

Thank you very much! 
It seems right now

**Comment by Violetit** [comment]
> use sel0[1] instead of sel0

 Hi @DiegoWangSys 
during the test process,  something goes wrong, the error was reported below:
```
10/16 00:49:29 ['/disk/tia/FCGF/dataset/threedmatch_test/7-scenes-redkitchen', '/disk/tia/FCGF/dataset/threedmatch_test/7-scenes-redkitchen-evaluation', '/disk/tia/FCGF/dataset/threedmatch_test/sun3d-home_at-home_at_scan1_2013_jan_1', '/disk/tia/FCGF/dataset/threedmatch_test/sun3d-home_at-home_at_scan1_2013_jan_1-evaluation', '/disk/tia/FCGF/dataset/threedmatch_test/sun3d-home_md-home_md_scan9_2012_sep_30', '/disk/tia/FCGF/dataset/threedmatch_test/sun3d-home_md-home_md_scan9_2012_sep_30-evaluation', '/disk/tia/FCGF/dataset/threedmatch_test/sun3d-hotel_uc-scan3', '/disk/tia/FCGF/dataset/threedmatch_test/sun3d-hotel_uc-scan3-evaluation', '/disk/tia/FCGF/dataset/threedmatch_test/sun3d-hotel_umd-maryland_hotel1', '/disk/tia/FCGF/dataset/threedmatch_test/sun3d-hotel_umd-maryland_hotel1-evaluation', '/disk/tia/FCGF/dataset/threedmatch_test/sun3d-hotel_umd-maryland_hotel3', '/disk/tia/FCGF/dataset/threedmatch_test/sun3d-hotel_umd-maryland_hotel3-evaluation', '/disk/tia/FCGF/dataset/threedmatch_test/sun3d-mit_76_studyroom-76-1studyroom2', '/disk/tia/FCGF/dataset/threedmatch_test/sun3d-mit_76_studyroom-76-1studyroom2-evaluation', '/disk/tia/FCGF/dataset/threedmatch_test/sun3d-mit_lab_hj-lab_hj_tea_nov_2_2012_scan1_erika', '/disk/tia/FCGF/dataset/threedmatch_test/sun3d-mit_lab_hj-lab_hj_tea_nov_2_2012_scan1_erika-evaluation']
Format = auto
Extension = ply
10/16 00:49:29 0 / 60: 7-scenes-redkitchen_000
Traceback (most recent call last):
  File "/home/ubuntu/.conda/envs/py3-fcgf/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/ubuntu/.conda/envs/py3-fcgf/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/disk/tia/FCGF/scripts/benchmark_3dmatch.py", line 257, in <module>
    device)
  File "/disk/tia/FCGF/scripts/benchmark_3dmatch.py", line 65, in extract_features_batch
    skip_check=True)
  File "/disk/tia/FCGF/util/misc.py", line 82, in extract_features
    coords = coords[inds]
IndexError: shape mismatch: indexing arrays could not be broadcast together with shapes (18977,3) (18977,) 
```
Cloud you please help me to solve this problem?
Thank you very much!

**Comment by chrischoy** [comment]
Fixed on the latest commit cc5b70008fd1282d19ee60f6c022d4b5d059f4bc.
